File: emulinker/src/main/java/org/emulinker/kaillera/controller/v086/protocol/Chat_Request.java
package org.emulinker.kaillera.controller.v086.protocol;

import com.google.auto.value.AutoValue;
import org.emulinker.kaillera.controller.messaging.MessageFormatException;

@AutoValue
public abstract class Chat_Request extends Chat {
  private static final String DESC = "Chat Request";
  private static final String USERNAME = "";

  public static AutoValue_Chat_Request create(int messageNumber, String message)
      throws MessageFormatException {
    V086Message.validateMessageNumber(messageNumber, DESC);
    return new AutoValue_Chat_Request(messageNumber, ID, DESC, USERNAME, message);
  }
}

File: emulinker/src/main/java/org/emulinker/kaillera/controller/v086/protocol/Chat_Request.kt
package org.emulinker.kaillera.controller.v086.protocol

import org.emulinker.kaillera.controller.messaging.MessageFormatException

data class Chat_Request
    @Throws(MessageFormatException::class)
    constructor(
        override val messageNumber: Int, override val message: String, override val username: String
    ) : Chat() {

  override val description = DESC
  override val messageId = ID

  init {
    validateMessageNumber(messageNumber, DESC)
  }

  companion object {
    private const val DESC = "Chat Request"
  }
}
